hnn_core/cell.py

```diff
@@ -137,26 +141,134 @@ def gid(self, gid):
         else:
             raise RuntimeError('Global ID for this cell already assigned!')
 
-    @abstractmethod
-    def get_sections(self):
-        """Get sections in a cell."""
-        pass
-
-    def create_soma(self):
-        """Create soma and set geometry."""
-        # make L_soma and diam_soma elements of self
-        # Used in shape_change() b/c func clobbers self.soma.L, self.soma.diam
-        soma_props = self.soma_props
-
-        self.L = soma_props['L']
-        self.diam = soma_props['diam']
-        self.pos = soma_props['pos']
+    def _set_biophysics(self, p_secs):
+        "Set the biophysics for the default Pyramidal cell."
+
+        # neuron syntax is used to set values for mechanisms
+        # sec.gbar_mech = x sets value of gbar for mech to x for all segs
+        # in a section. This method is significantly faster than using
+        # a for loop to iterate over all segments to set mech values
+
+        # Value depends on distance from the soma. Soma is set as
+        # origin by passing cell.soma as a sec argument to h.distance()
+        # Then iterate over segment nodes of dendritic sections
+        # and set gbar_ar depending on h.distance(seg.x), which returns
+        # distance from the soma to this point on the CURRENTLY ACCESSED
+        # SECTION!!!
+        h.distance(sec=self.soma)
+        for sec in self.sections:
+            sec_name = sec.name().split('_', 1)[1]
+            for mech_name, mech in p_secs[sec_name]['mechs'].items():
+                sec.insert(mech_name)
+                for attr, val in mech.items():
+                    if hasattr(val, '__call__'):
+                        sec.push()
+                        for seg in sec:
+                            setattr(seg, attr, val(h.distance(seg.x)))
+                        h.pop_section()
+                    else:
+                        setattr(sec, attr, val)
+
+    def _create_synapses(self, p_secs, p_syn):
+        """Create synapses."""
+        for sec_name in p_secs:
+            for receptor in p_secs[sec_name]['syns']:
+                sec_name_sanitized = sec_name.replace('_', '')
+                syn_key = f'{sec_name_sanitized}_{receptor}'
+                if sec_name == 'soma':
+                    seg = self.soma(0.5)
+                else:
+                    seg = self.dends[sec_name](0.5)
+                self.synapses[syn_key] = self.syn_create(
+                    seg, **p_syn[receptor])
+
+    def _create_sections(self, p_secs):
+        """Create soma and set geometry.
+
+        Notes
+        -----
+        By default neuron uses xy plane
+        for height and xz plane for depth. This is opposite for model as a
+        whole, but convention is followed in this function ease use of gui.
```

Comment:
```suggestion
        whole, but the convention is followed in this function 
        for ease use of with the gui.
```
Not sure I understand the connection with the gui

Reply:
I guess in the past the GUI had a visualization of the cells in 3D space. The z coordinate is along the trunk of the pyramidal cells (depth) but the Neuron convention is y coordinate. I just copy pasted the old comment. Feel free to suggest a better rephrasing!